### ISSUE TITLE
gdal broken because of cfitsio update

### ISSUE THREAD

**Issue by cpaulik** [conda-forge/gdal-feedstock#303]
```
conda create -n gdal-test python=2.7 gdal=2.4.*
````

leads to an environment with the following error:

```
from osgeo import gdal
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cpa/.pyenv/versions/miniconda2-4.3.30/envs/gdal-test/lib/python2.7/site-packages/osgeo/__init__.py", line 21, in <module>
    _gdal = swig_import_helper()
  File "/home/cpa/.pyenv/versions/miniconda2-4.3.30/envs/gdal-test/lib/python2.7/site-packages/osgeo/__init__.py", line 17, in swig_import_helper
    _mod = imp.load_module('_gdal', fp, pathname, description)
ImportError: libcfitsio.so.5: cannot open shared object file: No such file or directory
```

`conda install cfitsio=3.430` solves the problem?

Is the approach to rebuild or to pin cfitsio?

**Comment by ocefpaf** [comment]
We should add `cfitsio` to the pinning library. I'm looking into it. Hold on...

**Comment by ocefpaf** [comment]
Yep. There are a soname change. See https://abi-laboratory.pro/?view=timeline&l=cfitsio

**Comment by trey-stafford** [comment]
Seeing the same with python 3.7.3 and gdal v2.4.1, v2.4.2.

I tried installing cfitsio=3.430 as @cpaulik suggests, leading to a new error:
```
$  python -c "import gdal"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/trst2284/miniconda3/envs/gdal2/lib/python3.7/site-packages/gdal.py", line 2, in <module>
    from osgeo.gdal import deprecation_warn
  File "/home/trst2284/miniconda3/envs/gdal2/lib/python3.7/site-packages/osgeo/__init__.py", line 21, in <module>
    _gdal = swig_import_helper()
  File "/home/trst2284/miniconda3/envs/gdal2/lib/python3.7/site-packages/osgeo/__init__.py", line 17, in swig_import_helper
    _mod = imp.load_module('_gdal', fp, pathname, description)
  File "/home/trst2284/miniconda3/envs/gdal2/lib/python3.7/imp.py", line 242, in load_module
    return load_dynamic(name, filename, file)
  File "/home/trst2284/miniconda3/envs/gdal2/lib/python3.7/imp.py", line 342, in load_dynamic
    return _load(spec)
ImportError: libiconv.so.2: cannot open shared object file: No such file or directory
```

**Comment by ocefpaf** [comment]
@cpaulik when https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/246 and https://github.com/conda-forge/cfitsio-feedstock/pull/18 get merged we can re-render and rebuild `gdal`.

**Comment by trey-stafford** [comment]
To fix the `libiconv.so.2` problem after installing `cfitsio=3.430`, install `libstdcxx-ng=8.2.0`.
```
$ python -c "import gdal"
$ echo $?
0
```

**Comment by cpaulik** [comment]
Just tested it and it is of course fixed. 

Thanks @ocefpaf

**Comment by cpaulik** [comment]
I'm still facing issues because of this. 

Now some of our projects have trouble because of a dependency on both `gdal` and `h5py`.

e.g. 

`conda create -n gdal-test python=2.7 gdal h5py`

Now leads to the same problem because `gdal               conda-forge/linux-64::gdal-3.0.0-py27hf242f0b_3` is installed instead of the newest one.

What is the best approach?

**Comment by ocefpaf** [comment]
> What is the best approach?

Remove the old build :smile: 

I was waiting for the new package to land to do that but did not got to it yet. Hold on...

**Comment by ocefpaf** [comment]
I removed the old builds and the channel should update in ~50 min. Please let me know if you test it again.

**Comment by kmuehlbauer** [comment]
@ocefpaf That doesn't work out:

```
conda create -n gdal3 python=3.7 gdal=3* h5py
Collecting package metadata: done
Solving environment: failed

UnsatisfiableError: The following specifications were found to be in conflict:
  - gdal=3 -> libgdal==3.0.1=h0845e09_3 -> hdf5[version='>=1.10.5,<1.10.6.0a0']
  - h5py

```

**Comment by ocefpaf** [comment]
I'm playing the dependency whack-a-mole game now. Removing all the packages that got built with `cfitsio` without `run_exports` and rebuilding them. It may take a while though b/c I'm at a conference this week.

PS: @kmuehlbauer there is a dev version of `h5py` that uses latest `hdf5` and should work there.

**Comment by cpaulik** [comment]
Did you by any chance remove `gdal 2.4.x` builds which were built with `proj<6`? Because now some other builds are failing which depend on proj version 5.

**Comment by ocefpaf** [comment]
> Did you by any chance remove `gdal 2.4.x` builds which were built with `proj<6`? Because now some other builds are failing which depend on proj version 5.

We had to remove them but I'm slowly rebuilding them. For those who are already on `proj4 6` we have a consistent stack (including cartopy) for mscOS and Linux (no Windows yet). Both Windows stack with `proj4 6` and old builds of `gdal` with `proj4` 5 should appear (hopefully) later today.

**Comment by cpaulik** [comment]
Thanks. Great work.

**Comment by cpaulik** [comment]
I still have the following problem:

```
conda create -n test gdal=2.4 proj4=5.2  
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
Collecting package metadata (repodata.json): done
Solving environment: failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

  - gdal=2.4 -> python[version='>=3.6,<3.7.0a0'] -> openssl[version='>=1.1.1a,<1.1.2a'] -> ca-certificates
  - gdal=2.4 -> python[version='>=3.6,<3.7.0a0'] -> openssl[version='>=1.1.1a,<1.1.2a'] -> libgcc-ng[version='>=7.3.0'] -> _libgcc_mutex=[build=main]
  - gdal=2.4 -> python[version='>=3.6,<3.7.0a0'] -> tk[version='>=8.6.9,<8.7.0a0']
  - gdal=2.4 -> python[version='>=3.6,<3.7.0a0'] -> xz[version='>=5.2.4,<5.3.0a0']


Note that strict channel priority may have removed packages required for satisfiability.
```

I will try re-building my own conda packages with the latest gdal and proj to solve it. But `gdal 2.4` with `proj4 5` does not seem to work at the moment.

**Comment by ocefpaf** [comment]
Should be OK not. I fixed `libgdal` but missed `gdal` in the `2.4.1` `proj.4 5` PR.
